### PR TITLE
Add Keycloak gatekeeper chart

### PR DESCRIPTION
**Pull request by gabibbo97**
Keycloak gatekeeper version 4.6.0.Final

Signed-off-by: Giacomo Longo <gabibbo97@gmail.com>

#### What this PR does / why we need it:

This adds keycloak gatekeeper authenticating proxy

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [X] Variables are documented in the README.md

### CODE REVIEW

**Comment by monotek** [comment]
Please reopen.

File: stable/keycloak-gatekeeper/values.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  #  memory: 128Mi
+
+nodeSelector: {}
+
+tolerations: []
+
+affinity: {}
+
+# URL for OpenID autoconfiguration
+# On Keycloak <server>/auth/realm/<realm_name>
```
Comment (axdotl) [inline stable/keycloak-gatekeeper/values.yaml]: Has to be `realms` instead of `realm`
`<server>/auth/realms/<realm_name>`

**Comment by davidkarlsen** [comment]
@gabibbo97 Can you add RBAC please? https://github.com/helm/helm/blob/master/docs/chart_best_practices/rbac.md

File: stable/keycloak-gatekeeper/values.yaml (hunk 1 above)
Comment (gabibbo97) [reply stable/keycloak-gatekeeper/values.yaml]: Changed with the latest revision fb8f3f1

**Comment by gabibbo97** [comment]
@davidkarlsen Thanks for the review, I added a6db39455e64739157c04348f8558a4db01f1ee4 , let me know if I did correctly

**Comment by monotek** [comment]
Quick note: We're already using your chart for Grafana, Kibana & Prometheus and were able to completely drop basic auth :-)

**Comment by gabibbo97** [comment]
Well, now that there is the hub and after almost three months of waiting, I am going to host this inside my own private helm repository

**Comment by monotek** [comment]
Nice! :-)
Do you already have a link?
We did the same for fluentd-elasticsearch & graphite @ https://github.com/kiwigrid/helm-charts

**Comment by gabibbo97** [comment]
I am trying to set up a CI/CD process for my charts at the moment, the repo https://github.com/gabibbo97/charts is empty :blush: but will come later today with an update for gatekeeper to later versions

**Comment by monotek** [comment]
Feel free to copy our ci pipeline from: https://github.com/kiwigrid/helm-charts/tree/master/.circleci

Its heavily inspired by: https://github.com/helm/chart-testing/tree/master/examples/kind

I also could ask if we could host your chart in our repo, which is already incuded in helmhub.
I can tell you more tomorrow if you like...

**Comment by monotek** [comment]
@gabibbo97 
If you're interested to develop the gatekeeper chart together we're willing to host the chart in our repo and on hub.helm.sh.

At the moment i can promise fast reviews but no write access to the repo.

Our longterm goal is to implement the kubernetes ci robot (https://github.com/kubernetes/test-infra/tree/master/prow) also in our repo and give people in the OWNERS file of the charts the possibility to use the known /lgtm, /approve, /ok-to-test commands there.

**Comment by davidholsgrove** [comment]
Hi @gabibbo97 - is there any chance of this PR being reopened? Looks like a great addition, would prefer not to have to host this chart separately :)

**Comment by monotek** [comment]
Just use his own repo:

helm repo add gabibbo97 https://gabibbo97.github.io/charts

**Comment by gabibbo97** [comment]
> Hi @gabibbo97 - is there any chance of this PR being reopened? Looks like a great addition, would prefer not to have to host this chart separately :)

I opened the PR months ago, it was ignored, after ranting on slack it was reopened.
After one month of being ignored I decided that there was no interest at all and made my own chart repo because I still needed this chart.

Now my repo is available at hub.helm.sh

**Pull request closed without merging**